runtime/src/append_vec.rs

```diff
@@ -363,7 +363,7 @@ impl AppendVec {
             append_lock: Mutex::new(()),
             current_len: AtomicUsize::new(current_len),
             file_size,
-            remove_on_drop: true,
```

Comment:
It's probably good to have these `remove_on_drop` defaulting to false. But we do want to remove when we drop old append vecs that we've cleaned. So we need to set `remove_on_drop` to `true` then, otherwise we'll never delete any account state.

Reply:
^ I'd probably break this change out into a separate PR so that it's easier to verify.

Basically just make sure that we still delete cleaned appendvecs, but don't delete them when the application closes normally.

Reply:
OK, I will generate a separate PR to set the remove_on_drop flag to true when an appendvec becomes clean.  Thanks!

Reply:
In video discussion with Andrew we discussed the design to ensure that the unneeded files in the accounts/ are still deleted.

We will first create a to_be_deleted directory, then move all the /accounts/ files into that directory.  It should be cheap operation not costing much time.  Then at the unpacking stage, if any such file is referenced by the snapshot, then move that file back into accounts.  
After unpacking is done, the remaining files in the to_be_deleted/ directory will be deleted.

I will create a separate PR to implement this design.